tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/consolidatedAuth/ConsolidatedAuthUserReader.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.tasklist.webapp.security.consolidatedAuth;
+
+import static io.camunda.tasklist.webapp.security.TasklistProfileService.CONSOLIDATED_AUTH_PROFILE;
+
+import io.camunda.authentication.entity.CamundaPrincipal;
+import io.camunda.tasklist.webapp.dto.UserDTO;
+import io.camunda.tasklist.webapp.security.Permission;
+import io.camunda.tasklist.webapp.security.UserReader;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile(CONSOLIDATED_AUTH_PROFILE)
+public class ConsolidatedAuthUserReader implements UserReader {
+
+  @Override
+  public Optional<UserDTO> getCurrentUserBy(final Authentication authentication) {
+    if (authentication == null) {
+      return Optional.empty();
+    }
+    if (authentication.getPrincipal() instanceof final CamundaPrincipal camundaPrincipal) {
```

Comment:
:upside_down_face: instanceof is already a null check, no need for the extra check above